### DISCUSSION TITLE
Using immich-go with External Library

### DISCUSSION THREAD

**Discussion by srahimeen** [simulot/immich-go#470] [Q&A] [answered]
I have been manually backing up my Google Takeout, and immich-go seems like it does most of the things I use my custom scripts for, so I have been looking to migrate my workflow to use immich-go.

However, it seems like immich-go is built around taking Google Takeout downloads and uploading all the photos/videos to the Immich directory. I still use Google Photos, but I don't want to pay for  more Google One storage. I don't use Immich as a backup/sync solution, but more as a viewer/GUI for all the photos/videos I back up via Google Takeout.

My process is basically as follows:

1. Download Google Takeout a few times a year
2. Extract them in a `Google Photos` folder into my NAS
3. Have this `Google Photos` folder in Immich as an External Library
4. Delete old photos and videos from Google Photos to clear out storage space
5. Use Immich as a viewer for _all_ my photos and videos (which are on my NAS), while Google Photos only has _current_ photos and videos only

I keep a `Google Photos` folder on my NAS instead of using the Immich directory because I use other applications which use these photos/videos, so they need to exist outside of any single app's directory.

Because of the metadata issues with Google Takeout, I have to use custom scripts/exiftools to do the metadata corrections myself and then force Re-scan my entire external library. However, I don't have any scripts to actually create albums in Immich, which is something immich-go seems do have support for.

I wanted to see if its possible to use immich-go with my workflow where I have a separate `Google Photos` folder on my NAS with all my media and have that as an External Library in Immich.

I would ideally like to:

1. Run immich-go in my main `Google Photos` folder on my NAS to do the metadata corrections
2. Use immich-go to create albums in Immich corresponding to the albums folder
3. But do all that without uploading anything to the main Immich directory as I don't want duplicates of the photos/videos that are already in my `Google Photos` folder on my NAS. Just use the External Library to perform the operations without actually uploading anything.

Is such a workflow possible at the moment with immich-go? Or is it very coupled to _uploading_ to the Immich directory?

**Comment by simulot** [comment, marked as answer]
No, immich-go is designed to upload photos through the immich's API. 

Immich-go provides all you require, but using Immich's internal library. 
It manages the takeout duplicates transparently. 
If you have already the same photo in immich (same name, same date), it will add it to the same album is any. 

That's said, I'll work on the possibility to reuse the logic for solving takeout issues and place photos in a folder tree, with a nice XMP file to get accurate dates and GPS. But it's not ready yet

**Comment by srahimeen** [comment]
Thanks for the quick response! If I was all-in on the Immich internal library this would be the perfect utility for me.

If its possible to just use immich-go to do an in-place album creation and population (using the external library's files and no uploading), that would be perfect as I could just run immich-go at the end of my regular Immich import process.

**Reply by simulot** [reply]
I haven't try using it this way.
I suggest you to test it with a takeout limited to one or two albums with a few photos.
Let us know your findings
